### DISCUSSION TITLE
Questions about postprocessing when using Yolov5

### DISCUSSION THREAD

**Discussion by YoungjaeDev** [obss/sahi#640] [Q&A] [answered]
Hello
A demo test is conducted using Yolov5, and if I use the default parameter, it is wrapped in the `Autoshape` class and is forwarded
Then basically, it looks like doing `NMM` again after the included `NMS`, is this right?
I don't think there will be a problem even if I mix the two, but is there any problem in this area?

THANK YOU

**Comment by fcakyon** [comment, marked as answer]
@youngjae-avikus thanks for the question.

There are two use-case for SAHI:

a. don't perform sliced-inference:
  1. SAHI doesn't apply postprocess, only postprocess from yolov5/mmdet/detectron2 is performed.
 
b. perform sliced-inference:
  1. yolov5/mmdet etc performs prediction and applies postprocess for each single slice
  3. SAHI combines overlaping slice prediction results
  4. SAHI applies postprocess to combined results (to fix the duplicates coming from overlaps)
    
So detector postprocess and SAHI postprocess are different and both are required.

**Comment by YoungjaeDev** [comment]
@fcakyon 
In the case of YOLOV5, if I don't specify a specific parameter in the `postprocess` when you only do `FULL INFERENCE`, code only do NMS and don't do MERGE, right?

**Reply by fcakyon** [reply]
Yes un that case only default yolov5 postprocess is applied
